kinds/dashboard/dashboard_kind.cue

```diff
@@ -97,39 +105,72 @@ lineage: seqs: [
-					limit:    int64
+					// Only required/valid for the grafana datasource... 
+					// but code+tests is already dependin on it so hard to change
+					limit: int64
+					// Only required/valid for the grafana datasource... 
+					// but code+tests is already dependin on it so hard to change
 					matchAny: bool
+					// Only required/valid for the grafana datasource... 
+					// but code+tests is already dependin on it so hard to change
 					tags: [...string]
+					// Only required/valid for the grafana datasource... 
+					// but code+tests is already dependin on it so hard to change
 					type: string
+					... // datasource will stick thier raw DataQuery here
 				} @cuetsy(kind="interface") @grafanamaturity(NeedsExpertReview)
 
+				#AnnotationPanelFilter: {
+					// Should the specified panels be included or excluded
+					exclude?: bool | *false
+
+					// Panel IDs that should be included or excluded
+					ids: [...uint8]
+				}
+
 				// TODO docs
 				// FROM: AnnotationQuery in grafana-data/src/types/annotations.ts
 				#AnnotationQuery: {
-					// Datasource to use for annotation.
+					@grafana(TSVeneer="type")
+
+					// Name of annotation.
+					name: string
+
+					// TODO: Should be DataSourceRef
 					datasource: {
 						type?: string
 						uid?:  string
 					} @grafanamaturity(NeedsExpertReview)
 
-					// Whether annotation is enabled.
-					enable: bool | *true @grafanamaturity(NeedsExpertReview)
-					// Name of annotation.
-					name?:   string     @grafanamaturity(NeedsExpertReview)
-					builtIn: uint8 | *0 @grafanamaturity(NeedsExpertReview) // TODO should this be persisted at all?
-					// Whether to hide annotation.
-					hide?: bool | *false @grafanamaturity(NeedsExpertReview)
-					// Annotation icon color.
-					iconColor?: string                @grafanamaturity(NeedsExpertReview)
-					type:       string | *"dashboard" @grafanamaturity(NeedsExpertReview)
-					// Query for annotation data.
-					rawQuery?: string            @grafanamaturity(NeedsExpertReview)
-					showIn:    uint8 | *0        @grafanamaturity(NeedsExpertReview)
-					target?:   #AnnotationTarget @grafanamaturity(NeedsExpertReview)
+					// When enabled the annotation query is issued with every dashboard refresh
+					enable: bool | *true
+
+					// Annotation queries can be toggled on or off at the top of the dashboard.  
+					// When hide is true, the toggle is not shown in the dashboard.
+					hide?: bool | *false
+
+					// Color to use for the annotation event markers
+					iconColor: string
+
+					// Optionally   
+					filter?: #AnnotationPanelFilter
+
+					// TODO.. this should just be a normal query target
+					target?: #AnnotationTarget
+
+					// TODO -- this should not exist here, it is based on the --grafana-- datasource
+					type?: string @grafanamaturity(NeedsExpertReview)
+
+					// unless datasources have migrated to the target+mapping,
+					// thhey just spread their query into the base object :(
+					...
```

Comment:
@sdboyer is this the accepted choice for "or anything else" here?

Reply:
Yep, if we have to accept unknown fields being spilled in here as valid, this is how to mark it.

It may be possible to not do this if it's possible for us to know at schema-time whether the datasource plugin in question has done the "target+mapping migration" you mention in the comment (idk what that is). That would be analogous to how if we encounter a panel that's from a plugin for which we have no composable kind, we let the `options` field be anything/unrestricted